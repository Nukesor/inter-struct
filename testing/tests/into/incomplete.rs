use inter_struct::prelude::*;

/// This should crash, since the generated initializer for IntoStruct will be incomplete.
#[derive(StructInto)]
#[struct_into("crate::IntoStruct")]
pub struct FromStruct {
    pub normal: String,
}

pub struct IntoStruct {
    pub normal: String,
    pub optional: Option<String>,
}

fn main() {}
