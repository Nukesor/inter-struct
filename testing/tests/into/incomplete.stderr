error[E0063]: missing field `optional` in initializer of `IntoStruct`
 --> tests/into/incomplete.rs:5:15
  |
5 | #[struct_into("crate::IntoStruct")]
  |               ^^^^^^^^^^^^^^^^^^^ missing `optional`
